ReactiveCocoa/Swift/Signal.swift

```diff
@@ -86,15 +133,18 @@ public final class Signal<T, E: ErrorType> {
 	public func observe<S: SinkType where S.Element == Event<T, E>>(observer: S) -> Disposable? {
 		let sink = Observer(observer)
 
-		lock.lock()
-		let token = self.observers?.insert(sink)
-		lock.unlock()
+		var token: RemovalToken?
+		atomicObservers.modify { (var observers) in
```

Comment:
Didn't there used to be a version of `modify` that allowed you to pass-thru secondary data. Is it worth bringing that back?